File: budget.c
#include <stdio.h>

int main(void){
    float income, rent, utilities, groceries, transportation, savings, spent, expenses;
    float prent, pgroceries, ptransportation, psavings, pexpenses, putilities;
    printf("this is a budget calculator.\n How much money do you make a month? \n");
    scanf("%f", &income);
    printf("How much do your utilities cost? \n");
    scanf("%f", &utilities);
    printf("How much does your groceries cost? \n");
    scanf("%f", &groceries);
    printf("How much do your transportation cost? \n");
    scanf("%f", &transportation);
    printf("How much do your rent cost? \n");
    scanf("%f", &rent);
    savings = income *.2;
    expenses = rent + utilities + groceries + transportation;
    spent = income - expenses - savings;
    printf("you make %.2f\n", income);
    printf("your expenses are $%.2f\n", expenses);
    printf("your savings are $%.2f\n", savings);
    printf("your spending money is $%.2f\n", spent);
    prent = rent/income * 100;
    pgroceries = groceries/income * 100;
    ptransportation = transportation/income * 100;
    psavings = savings/income * 100;
    pexpenses = expenses/income * 100;
    printf("your rent is %d%% of your income.\n", (int) prent);
    printf("your utilities are %d%% of your income.\n", (int) putilities);
    return 0;
}

File: budget.py
print("This is a finacial calculator")
income = input("how much do you make a month?:\n" )
rent = input("how much is your rent?:\n")
utilities=input("how much is your utilities?:\n")
groceries=input("how much do your groceries cost?:\n")
transp=input("what does your transportation cost?:\n")
print("you make", income)
savings = float(income) * float(0.2)
expenses = (float(rent) + float(utilities) + float(groceries) + float(transp))
spending = float(income)-float(expenses)-float(savings)
print("your expenses are", expenses)
print("your savings is", savings)
print("your spending money is", spending)
prent=(float(rent)/float(income))*100
putilities=(float(utilities)/float(income))*100
pgroceries=(float(groceries)/float(income))*100
ptransp=(float(transp)/float(income))*100
print("your rent takes up this percent of your income;", prent)
print("your utilities take up this percent of your income;", putilities)
print("your groceries take up this percent of your income;", pgroceries)
print("your transportation needs take up this percent of your income;", ptransp)